boards: xtensa: Rename WeAct boards for consistency

heronet opened the pull request:
Update board.yml to use a more consistent name and update compatible string to 'weact,esp32s3-b' instead of 'espressif,esp32s3'.

marekmatej commented on boards/weact/weact_esp32s3_b/board.yml:
```diff
@@ -1,6 +1,6 @@
 board:
   name: weact_esp32s3_b
-  full_name: WeAct Studio ESP32-S3-B
```
the full name can stay as is.

marekmatej replied:
Ok, got your point. There are 2 boards + 1shield with this redundancy in name. Can you fix them all?

heronet replied:
Yes I can. Do you think it will be better to do the other 2 in a seperate commit?

heronet replied:
Can you take another look? sorry for the delay

heronet committed:
boards: xtensa: Rename WeAct boards for consistency

Update weact_esp32s3_b, weact_esp32c6_mini, weact_ov2640_cam_module
to have more consistent naming.

Signed-off-by: Siratul Islam <email@sirat.me>
